src/Entries/Entry.php

```diff
@@ -17,8 +17,8 @@ public static function fromModel(Model $model)
             ->origin($model->origin_id)
             ->locale($model->site)
             ->slug($model->slug)
-            ->date($model->date)
             ->collection($model->collection)
+            ->date($model->date)
```

Comment:
You'll wanna move this a bit lower and wrap in a condition.

e.g. 

```
if ($model->date) {
  $entry->date($model->date);
}
```

If you try to set `date()` on a non-dated collection entry, you'll get the error mentioned in https://github.com/statamic/cms/issues/7959

I'm not 100% sure, but I think `$model->date` would only ever exist if the collection is dated, so just checking if the attribute isn't null is enough.